straxen/plugins/peak_processing.py

```diff
@@ -332,97 +334,316 @@ def find_n_competing(peaks, windows, fraction):
+                 help='Search for ambience in this radius [cm]'),
+)
+class PeakAmbience(strax.OverlapWindowPlugin):
+    """
+    Calculate Ambience of peaks.
+    Features are the number of lonehits, small S0, S1, S2 in a time window before peaks,
+    and the number of small S2 in circle near the S2 peak in a time window.
+    References:
+        * v0.0.4 reference: xenon:xenonnt:ac:prediction:shadow_ambience
+    """
+    __version__ = '0.0.4'
+    depends_on = ('lone_hits', 'peak_basics', 'peak_positions')
+    provides = 'peak_ambience'
+    data_kind = 'peaks'
+    save_when = strax.SaveWhen.EXPLICIT
+
+    def setup(self):
+        self.ambience_divide_time = self.config['ambience_divide_time']
+        self.ambience_divide_space = self.config['ambience_divide_space']
+        self.time_window_ambience = self.config['time_window_ambience']
+        self.ambience_area_parameters = self.config['ambience_area_parameters']
+        self.ambience_radius = self.config['ambience_radius']
+
+    def get_window_size(self):
+        return 3 * self.time_window_ambience
+
+    @property
+    def origin_dtype(self):
+        return ['lh_before', 's0_before', 's1_before', 's2_before', 's2_near']
+
+    def infer_dtype(self):
+        dtype = []
+        for ambience in self.origin_dtype:
+            dtype += [(('Number of small ' + ' '.join(ambience.split('_')) + ' a peak', f'n_{ambience}'), np.int16), 
+                      (('Area sum of small ' + ' '.join(ambience.split('_')) + ' a peak', f's_{ambience}'), np.float32)]
+        dtype += strax.time_fields
+        return dtype
+
+    def compute(self, lone_hits, peaks):
+        return self.compute_ambience(lone_hits, peaks, peaks)
+
+    def compute_ambience(self, lone_hits, peaks, current_peak):
+        # 1. Initialization
+        result = np.zeros(len(current_peak), self.dtype)
+        num_array = np.zeros(len(current_peak), dtype=np.int16)
+        sum_array = np.zeros(len(current_peak), dtype=np.float32)
+
+        # 2. Define time window for each peak, we will find small peaks & lone hits within these time windows
+        roi = np.zeros(len(current_peak), dtype=strax.time_fields)
+        roi['time'] = current_peak['center_time'] - self.time_window_ambience
+        roi['endtime'] = current_peak['center_time']
+
+        # 3. Calculate number and area sum of lonehits before a peak
+        touching_windows = strax.touching_windows(lone_hits, roi)
+        # Calculating ambience
+        self.lonehits_ambience(current_peak, lone_hits, touching_windows, num_array, sum_array, self.ambience_divide_time)
+        result['n_lh_before'] = num_array
+        result['s_lh_before'] = sum_array
+
+        # 4. Calculate number and area sum of small S0, S1, S2 before a peak
+        radius = -1
+        for stype, area, type_str in zip([0, 1, 2], self.ambience_area_parameters, ['s0_before', 's1_before', 's2_before']):
+            num_array = np.zeros(len(current_peak), dtype=np.int16)
```

Comment:
You do not need the type_str do you? You can also do in the end `result[f'n_s{stype}_before']` etc.